pkgs/applications/misc/treedome/default.nix

```diff
@@ -0,0 +1,115 @@
+{ lib
+, cargo-tauri
+, cmake
+, dbus
+, fetchFromGitLab
+, fetchYarnDeps
+, freetype
+, gsettings-desktop-schemas
+, gtk3
+, libsoup
+, mkYarnPackage
+, openssl
+, pkg-config
+, rustPlatform
+, webkitgtk
+, wrapGAppsHook
+, sqlite
+}:
+
+let
+  pname = "treedome";
+  version = "0.3.2";
+
+  src = fetchFromGitLab {
+    owner = "treedome";
+    repo = "treedome";
+    rev = version;
+    sha256 = "sha256-0f32xa5s43s3amna0l9aiwcjr98679hkrsb7fxg3s1n9hr8vdz4z";
+    fetchSubmodules = true;
+  };
+
+  frontend-build = mkYarnPackage {
+    inherit version src;
+    pname = "treedome-ui";
+
+    offlineCache = fetchYarnDeps {
+      yarnLock = "${src}/yarn.lock";
+      sha256 = "sha256-rV5jKKnbMutaG5o8gRKgs/uoKwbIkxAPIcx6VWG7mm4=";
+    };
+
+    packageJSON = "${src}/package.json";
+
+    configurePhase = ''
+      ln -s $node_modules node_modules
+    '';
+
+    buildPhase = ''
+      export HOME=$(mktemp -d)
+      yarn --offline run build
+
+      mkdir -p $out/dist
+      cp -r dist/** $out/dist
+    '';
+
+    distPhase = "true";
+    dontInstall = true;
+  };
+in
+rustPlatform.buildRustPackage {
+  inherit version pname src;
+  sourceRoot = "src-tauri";
+
+  # essential in NVIDIA + compositor https://github.com/NixOS/nixpkgs/issues/212064#issuecomment-1400202079
+  WEBKIT_DISABLE_COMPOSITING_MODE = 1;
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "fix-path-env-0.0.0" = "sha256-ewE3CwqLC8dvi94UrQsWbp0mjmrzEJIGPDYtdmQ/sGs=";
+    };
+  };
+
+  # copy the frontend static resources to final build directory
+  # Also modify tauri.conf.json so that it expects the resources at the new location
+  postPatch = ''
+    mkdir -p dist
+    cp -R ${frontend-build}/dist/** dist
+
+    ls -al
+    substituteInPlace ./tauri.conf.json --replace '"distDir": "../dist",' '"distDir": "dist",'
+    substituteInPlace ./tauri.conf.json --replace '"beforeBuildCommand": "yarn run build",' '"beforeBuildCommand": "",'
+  '';
+
+  nativeBuildInputs = [ cmake pkg-config cargo-tauri wrapGAppsHook ];
+  buildInputs = [ dbus openssl freetype libsoup gtk3 webkitgtk gsettings-desktop-schemas sqlite ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    cargo tauri build
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin/
+    mkdir -p $out/share/
+
+    cp target/release/bundle/deb/treedome_0.0.0_amd64/data/usr/bin/treedome $out/bin/treedome
+    cp -R target/release/bundle/deb/treedome_0.0.0_amd64/data/usr/share/** $out/share/
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A local-first, encrypted, note taking application with tree-like structures, all written and saved in markdown.";
```

Comment:
The description should not include a period at the end. See https://nixos.org/manual/nixpkgs/stable/#var-meta-description